### PR TITLE
rstein - fixes

### PR DESCRIPTION
**Pull request by RalphSteinhagen**
follow-up fixes in response to 11.1.2 release and miscellaneous Coverity and LGTM fixes

### CODE REVIEW

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ chartfx-dataset/src/main/java/de/gsi/dataset/event/EventSource.java  2
+ chartfx-dataset/src/main/java/de/gsi/dataset/spi/ListErrorDataSet.java  4
+ chartfx-dataset/src/main/java/de/gsi/dataset/spi/ListDataSet.java  5
         

Coverage decreased per file
===========================
- chartfx-dataset/src/main/java/de/gsi/dataset/spi/DataSetBuilder.java  -1
         

Complexity increasing per file
==============================
- chartfx-samples/src/main/java/de/gsi/chart/samples/MountainRangeRendererSample.java  6
- chartfx-samples/src/main/java/de/gsi/math/samples/WaveletDenoising.java  3
- chartfx-samples/src/main/java/de/gsi/chart/samples/ContourChartSample.java  6
         

Clones added
============
- chartfx-samples/src/main/java/de/gsi/chart/samples/MountainRangeRendererSample.java  1
- chartfx-samples/src/main/java/de/gsi/chart/samples/ContourChartSample.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=4977386&bid=16440506)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-samples/src/main/java/de/gsi/chart/samples/ContourChartSample.java  6
- chartfx-samples/src/main/java/de/gsi/math/samples/WaveletDenoising.java  3
- chartfx-samples/src/main/java/de/gsi/chart/samples/MountainRangeRendererSample.java  6
         

Clones added
============
- chartfx-samples/src/main/java/de/gsi/chart/samples/ContourChartSample.java  1
- chartfx-samples/src/main/java/de/gsi/chart/samples/MountainRangeRendererSample.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=4977386&bid=16440506)

File: chartfx-chart/src/main/java/de/gsi/chart/Chart.java [hunk 1]
```diff
@@ -854,17 +854,17 @@ public final BooleanProperty legendVisibleProperty() {
...
-            return state;
+            return false;
         }
+        final Node axisNode = (Node) node;
         // remove axis from all axis panes
         for (final Side side : Side.values()) {
-            if (getAxesPane(side).getChildren().remove(node)) {
-                state = true;
+            if (getAxesPane(side).getChildren().remove(axisNode)) {
+                return true;
```
[the remark this answers is not in the document]
Comment (RalphSteinhagen) [reply chartfx-chart/src/main/java/de/gsi/chart/Chart.java]: The JavaFX rendering tree structure only allows unique nodes (ie. any node can be added only once to the tree) and that by the Chart design can thus only exclusively added to one of the four panes (or not at all). The axisNode.getParent().getChildrenUnmodifiable() (for non-null parent) does not allow to remove the axis element.

File: chartfx-samples/src/main/java/de/gsi/chart/samples/ContourChartSample.java [hunk 2]
```diff
@@ -145,11 +145,19 @@ public DataSet3D readImage() {
             // final BufferedReader reader = new BufferedReader(new
             // InputStreamReader(
             // ContourChartSampleReference.class.getResourceAsStream("./testdata/image.txt")));
-            reader.readLine();
+            @SuppressWarnings("unused")
+            String skipLine;
+            if ((skipLine = reader.readLine()) == null) {
```
[the remark this answers is not in the document]
Comment (wirew0rm) [reply chartfx-samples/src/main/java/de/gsi/chart/samples/ContourChartSample.java]: fixes warnings for readLine() with unused result

**Pull request merged**